azurerm/data_source_subnet.go

```diff
@@ -92,6 +97,10 @@ func dataSourceArmSubnetRead(d *schema.ResourceData, meta interface{}) error {
 	if props := resp.SubnetPropertiesFormat; props != nil {
 		d.Set("address_prefix", props.AddressPrefix)
 
+		if p := props.PrivateLinkServiceNetworkPolicies; p != nil {
+			d.Set("disable_private_link_service_network_policies", *p == "Disabled")
```

Comment:
By default this is set to "enabled" meaning that the NSG will control traffic security for all network traffic as normal however, for private link endpoints and private link services this needs to be set to "disabled" thus turning off all NSG routing/rules and allowing the traffic to flow through the private link or service. This is a little weird because by enabling you are effectively removing the NSG control over the network traffic. NSG is not supported on private endpoints/services. While subnets containing the private endpoint can have NSG associated with it, the rules will not be effective on traffic processed by the private endpoint. You must have network policies enforcement disabled to deploy private endpoints/services in a subnet. NSG is still enforced on other workloads hosted on the same subnet. Routes on any client subnet will be using an /32 prefix, changing the default routing behavior requires a similar UDR

Reply:
But technically you are enabling the private link service on the endpoint =/ it is really unfortunate naming

Reply:
given this is only applicable to Private Link Network Policies  - would it'd be worth making this:

```suggestion
			d.Set("enforce_private_link_network_policies", !strings.EqualsFold("Disabled", *p))
```

as a side-note it seems strange that the API exposes this internal behaviour at all?